src/translator/sentence_ranges.cpp

```diff
@@ -6,48 +6,44 @@ namespace marian {
 namespace bergamot {
 
 void Annotation::addSentence(std::vector<ByteRange> &sentence) {
-  size_t size = flatByteRanges_.size();
   flatByteRanges_.insert(std::end(flatByteRanges_), std::begin(sentence),
                          std::end(sentence));
-  sentenceBeginIds_.push_back(size);
+  size_t size = flatByteRanges_.size();
+  sentenceEndIds_.push_back(size);
 }
 
 size_t Annotation::numWords(size_t sentenceIdx) const {
-  auto terminals = sentenceTerminalIds(sentenceIdx);
-  return terminals.second - terminals.first + 1;
-}
-
-std::pair<size_t, size_t>
-Annotation::sentenceTerminalIds(size_t sentenceIdx) const {
   size_t bosId, eosId;
-  bosId = sentenceBeginIds_[sentenceIdx];
-  eosId = sentenceIdx + 1 < numSentences()
-              ? sentenceBeginIds_[sentenceIdx + 1] - 1
-              : flatByteRanges_.size() - 1;
-
-  // Out of bound checks.
-  assert(bosId < flatByteRanges_.size());
-  assert(eosId < flatByteRanges_.size());
-  return std::make_pair(bosId, eosId);
-}
-
-std::pair<ByteRange, ByteRange>
-Annotation::sentenceTerminals(size_t sentenceIdx) const {
-  auto terminals = sentenceTerminalIds(sentenceIdx);
-  return std::make_pair(flatByteRanges_[terminals.first],
-                        flatByteRanges_[terminals.second]);
+  bosId = sentenceEndIds_[sentenceIdx]; // Half interval, so;
+  eosId = sentenceEndIds_[sentenceIdx + 1];
+  // Difference between eosId and bosId is the number of words.
+  return eosId - bosId;
 }
 
 ByteRange Annotation::sentence(size_t sentenceIdx) const {
-  auto terminals = sentenceTerminals(sentenceIdx);
-  return (ByteRange){terminals.first.begin, terminals.second.end};
+  size_t bosId, eosId;
```

Comment:
If a user wants information for the very 1st sentence of the text, what should be provided as `sentenceIdx`? 0 or 1? Asking because the substring `Idx` of `sentenceIdx` can be interpreted that it starts from 0 and it is a valid value.

It makes sense to add something about the acceptable values for `sentenceIdx` parameter in the API description and clarify this 👍

Reply:
All public APIs should use 0 indexing.

Reply:
Thanks for clarifying. It means for the very 1st sentence of the text, sentenceIdx should be 0. Adding a line in API description just to clear this ambiguity would be helpful 👍

Reply:
@jerinphilip I don't know whether this PR is the right place for this change or you want to do it in a separate PR. But just mentioning that for 1st sentence/word, `sentenceIdx`/`wordIdx` parameter should be 0 in the public member functions of `Annotation` and `AnnotatedText` classes would be helpful.

I don't want to block this PR just because of this change. Rest of the PR looks good to me 👍

Reply:
I'll update the doxygen docs for API along with this PR, there seems to be bigger problems than this at the current stage which kpu's sending me at.